backend/src/schema/resolvers/posts.spec.js

```diff
@@ -91,44 +91,63 @@ afterEach(async () => {
 })
 
 describe('Post', () => {
-  const postQueryFilteredByCategories = gql`
-    query Post($filter: _PostFilter) {
-      Post(filter: $filter) {
-        id
-        categories {
-          id
-        }
-      }
-    }
-  `
-
-  const postQueryFilteredByEmotions = gql`
-    query Post($filter: _PostFilter) {
-      Post(filter: $filter) {
-        id
-        emotions {
-          emotion
-        }
-      }
-    }
-  `
-
   describe('can be filtered', () => {
-    let post31, post32
+    let followedUser, happyPost, cryPost
     beforeEach(async () => {
-      ;[post31, post32] = await Promise.all([
-        factory.create('Post', { id: 'p31', categoryIds: ['cat4'] }),
-        factory.create('Post', { id: 'p32', categoryIds: ['cat15'] }),
-        factory.create('Post', { id: 'p33', categoryIds: ['cat9'] }),
+      ;[followedUser] = await Promise.all([
+        factory.create('User', {
+          id: 'followed-by-me',
+          email: 'followed@example.org',
+          name: 'Followed User',
+          password: '1234',
+        }),
+      ])
+      ;[happyPost, cryPost] = await Promise.all([
+        factory.create('Post', { id: 'happy-post', categoryIds: ['cat4'] }),
+        factory.create('Post', { id: 'cry-post', categoryIds: ['cat15'] }),
+        factory.create('Post', {
+          id: 'post-by-followed-user',
+          categoryIds: ['cat9'],
+          author: followedUser,
+        }),
       ])
     })
 
+    describe('no filter', () => {
+      it('returns all posts', async () => {
+        const postQueryNoFilters = gql`
+          query Post($filter: _PostFilter) {
+            Post(filter: $filter) {
+              id
+            }
+          }
+        `
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-11T09:31:21Z" title="Wednesday, September 11th 2019, 11:31:21 am +02:00">Sep 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -91,44 +91,63 @@ afterEach(async () => {
 })
 
 describe('Post', () => {
-  const postQueryFilteredByCategories = gql`
-    query Post($filter: _PostFilter) {
-      Post(filter: $filter) {
-        id
-        categories {
-          id
-        }
-      }
-    }
-  `
-
-  const postQueryFilteredByEmotions = gql`
-    query Post($filter: _PostFilter) {
-      Post(filter: $filter) {
-        id
-        emotions {
-          emotion
-        }
-      }
-    }
-  `
-
   describe('can be filtered', () => {
-    let post31, post32
+    let followedUser, happyPost, cryPost
     beforeEach(async () => {
-      ;[post31, post32] = await Promise.all([
-        factory.create('Post', { id: 'p31', categoryIds: ['cat4'] }),
-        factory.create('Post', { id: 'p32', categoryIds: ['cat15'] }),
-        factory.create('Post', { id: 'p33', categoryIds: ['cat9'] }),
+      ;[followedUser] = await Promise.all([
+        factory.create('User', {
+          id: 'followed-by-me',
+          email: 'followed@example.org',
+          name: 'Followed User',
+          password: '1234',
+        }),
+      ])
+      ;[happyPost, cryPost] = await Promise.all([
+        factory.create('Post', { id: 'happy-post', categoryIds: ['cat4'] }),
+        factory.create('Post', { id: 'cry-post', categoryIds: ['cat15'] }),
+        factory.create('Post', {
+          id: 'post-by-followed-user',
+          categoryIds: ['cat9'],
+          author: followedUser,
+        }),
       ])
     })
 
+    describe('no filter', () => {
+      it('returns all posts', async () => {
+        const postQueryNoFilters = gql`
+          query Post($filter: _PostFilter) {
+            Post(filter: $filter) {
+              id
+            }
+          }
+        `
```


I believe it helps to avoid false negatives (test does not fail although there is a bug) if you use the very same graphql query throughout the scope. `await exect(...).resolves.toMatchObject(...)` should be lenient enough about any additional data in the `actual` response.